service/src/main/java/bio/terra/drshub/util/AccessMethodUtils.java

```diff
@@ -0,0 +1,58 @@
+package bio.terra.drshub.util;
+
+import static org.apache.commons.lang3.ObjectUtils.isEmpty;
+
+import bio.terra.drshub.config.DrsProvider;
+import bio.terra.drshub.generated.model.RequestObject.CloudPlatformEnum;
+import io.github.ga4gh.drs.model.AccessMethod;
+import io.github.ga4gh.drs.model.DrsObject;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Component
+@Slf4j
+public class AccessMethodUtils {
+  public static Optional<AccessMethod> getAccessMethod(
+      DrsObject drsResponse, DrsProvider drsProvider, CloudPlatformEnum cloudPlatform) {
+    Optional<AccessMethod> accessMethod = Optional.empty();
+    if (!isEmpty(drsResponse)) {
+      List<AccessMethod> accessMethods = getAccessMethods(drsResponse, drsProvider);
+      if (cloudPlatform != null) {
+        accessMethod = getAccessMethodForCloud(accessMethods, cloudPlatform);
+      }
+      // if there is no access method matching the cloudPlatform, or
+      // if the cloudPlatform was not specified, return a different access method
+      if (accessMethod.isEmpty() && !accessMethods.isEmpty()) {
+        accessMethod = Optional.of(accessMethods.get(0));
+      }
+    }
+    return accessMethod;
+  }
+
+  public static Optional<AccessMethod> getAccessMethodForCloud(
+      List<AccessMethod> accessMethods, CloudPlatformEnum cloudPlatform) {
+    Predicate<AccessMethod> filter;
+    if (cloudPlatform.equals(CloudPlatformEnum.AZURE)) {
+      filter = m -> m.getAccessId() != null && m.getAccessId().startsWith("az");
```

Comment:
We should put a comment that this is TDR behavior.  Not every DRS provider will prefix their access methods with `az`

Reply:
I'm a bit uneasy about baking in provider-specific handling into code rather than config, but I'm not sure what the clean alternative is.

Reply:
Yeah, unfortunately, there's nothing in the DRS spec to explicitly return the cloud type...